flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/DataFormatConverters.java

```diff
@@ -281,6 +288,28 @@ public static DataFormatConverter getConverterForDataType(DataType originDataTyp
 		return ps;
 	}
 
+	private static int getDateTimePrecision(LogicalType logicalType) {
+		if (logicalType instanceof TimestampType) {
+			TimestampType dt = (TimestampType) logicalType;
+			return dt.getPrecision();
+		} else {
+			TypeInformation typeInfo = ((LegacyTypeInformationType) logicalType).getTypeInformation();
+			if (typeInfo instanceof LegacyTimestampTypeInfo) {
+				LegacyTimestampTypeInfo dt = (LegacyTimestampTypeInfo) typeInfo;
+				return dt.getPrecision();
+			} else if (typeInfo instanceof LegacyLocalDateTimeTypeInfo) {
+				LegacyLocalDateTimeTypeInfo dt = (LegacyLocalDateTimeTypeInfo) typeInfo;
+				return dt.getPrecision();
+			} else if (typeInfo instanceof SqlTimeTypeInfo) {
+				return 3;
```

Comment:
Let's remove `SqlTimeTypeInfo` and `LocalTimeTypeInfo`, precision 3 is a temporary solution, and should not reach here.

Reply:
The SqlTimeTypeInfo and LocalTimeTypeInfo are mapping to Timestamp(3) in LegacyTypeInfoDataTypeConverter now, should we keep consistent with that?

Reply:
OK, we follow the definition in TimestampType, using 6 as the default precision.